src/routes/chat/[id]/+page.svelte

```diff
@@ -28,15 +28,40 @@
 	let autoscroll = true
 
 	beforeUpdate(() => {
-		autoscroll = div && div.offsetHeight + div.scrollTop > div.scrollHeight - 20
+		if (browser && window) {
+			let body = document.body
+			let html = document.documentElement
+
+			let height = Math.max(
+				body.scrollHeight,
+				body.offsetHeight,
+				html.clientHeight,
+				html.scrollHeight,
+				html.offsetHeight,
+			)
```

Comment:
```suggestion
		if (browser && window && dev) {
			const height = div.scrollHeight
```

This works for me on all browsers and to me the code is more understandable and as far as I understand the previous version it does the same thing.